src/net/sourceforge/kolmafia/persistence/DebugDatabase.java

```diff
@@ -2250,9 +2255,8 @@ private static void loadScrapeData(final StringArray array, final String fileNam
             currentHTML.append(currentLine);
             currentHTML.append(KoLConstants.LINE_BREAK);
           } while (!currentLine.equals("</html>"));
-
-          if (array.get(currentId).equals("")) {
-            array.set(currentId, currentHTML.toString());
+          if (stringMap.get(currentId).isEmpty() || stringMap.get(currentId) == null) {
```

Comment:
Oh yeah!

> One comment on nulls.
> 
> A separate comment on the behaviour of the original arrays: if you `set` beyond the bounds of the array, it would populate the intermediate values with the default (either 0 or ""). If you then iterated over or converted to an array, it would keep the zeroes (I believe). I can't find anything that did this, but it's something to watch out for.

Yeah I have tried to keep an eye on this - its why I use Map with getOrDefault where that feature may have been used, otherwise using a List